### PR TITLE
feat: idp/rp test apps

### PR DESCRIPTION
**Pull request by dopry**
Implement a minimal testing IDP and RP for maintainers.

There is a single Application configured in the IDP for the RP sample application it used the OIDC Authorization + PKCE flow.

This is a meant to be a starting point for building out further test scenarios.

applies to #1266

## Description of the Change

implements a test IDP using DOT and RP as a pre-cursor to developing playwright E2E tests. 

## Checklist

- [X] PR only contains one change (considered splitting up PR)
- [N/A] unit-test added
- [X] documentation updated
- [X] `CHANGELOG.md` updated (only for user relevant changes)
- [X] author name in `AUTHORS`

### CODE REVIEW

**Comment by tonial** [comment]
I'll look into this PR on Friday.

**Review by tonial — APPROVED** [review]
I think this test app is a great addition to this project.

I've never used svelte before so I don't really feel qualified to judge this part of this PR, but it works and it's easy to use, so I think it should be enough for anyone.

I think the svelte part would benefit from enhancements in the future to be able to test multiple scenarios (with or without PKCE, using basic auth or passing client_id + client_secret, ...), but it should not slow down this change.

File: tests/app/idp/idp/settings.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+STATIC_URL = "static/"
+
+# Default primary key field type
+# https://docs.djangoproject.com/en/4.2/ref/settings/#default-auto-field
+
+DEFAULT_AUTO_FIELD = "django.db.models.BigAutoField"
+
+OAUTH2_PROVIDER = {
+    "OIDC_ENABLED": True,
+    "OIDC_RP_INITIATED_LOGOUT_ENABLED": True,
```
[the remark this answers is not in the document]
Comment (dopry) [reply tests/app/idp/idp/settings.py:131]: In what scenarios would the user need to override these settings? They're set to work with the RP. The idea is that the two just work together and the user doesn't need to do anything.

File: tests/app/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+### Development Tasks
+
+* starting up the idp
+
+  ```bash
+  cd tests/app/idp
+  # create a virtual env if that is something you do
+  python manage.py migrate
+  python manage.py loaddata fixtures/seed.json
+  python manage.py run server
```
[the remark this answers is not in the document]
Comment (dopry) [reply tests/app/README.md]: Change pushed.

File: tests/app/idp/idp/settings.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tonial) [reply tests/app/idp/idp/settings.py:131]: If the goal of this test app is to allow developers to easily test new features or fixes on DOT, then they will probably need to change the settings.
As an example, if we wanted to change when the logout prompt display is mandatory we would have to add  `"OIDC_RP_INITIATED_LOGOUT_ALWAYS_PROMPT": False` here.

[the remark this answers is not in the document]
Comment (dopry) [reply tests/app/idp/idp/settings.py:131]: @tonial,  I feel like developers working on DOT and Django should know how to update settings.py to run a test scenario. I feel like if we implemented an alternative settings approach like django-environ or dotenv that we would need to document it then.  Would you be comfortable with me creating a new issue to add django-environ that I'll take ownership of implementing?

@n2ygk nice. The IDP here is pretty minimal, would you want to merge some of your work into this or maintain it as a standalone example implementation?

[the remark this answers is not in the document]
Comment (tonial) [reply tests/app/idp/idp/settings.py:131]: @dopry your are right, no need to add complexity here any django and DOT developer will be able to edit the settings.py file

**Pull request merged**